utils/filter.c

```diff
@@ -252,7 +252,13 @@ static void add_arg_spec(struct list_head *arg_list, struct uftrace_arg_spec *ar
 void update_trigger(struct uftrace_filter *filter, struct uftrace_trigger *tr, bool exact_match)
 {
 	if (tr->flags & TRIGGER_FL_CLEAR) {
-		filter->trigger.flags &= ~tr->flags;
+		/* No new flag other than CLEAR means delete all registered triggers */
+		if (tr->flags == TRIGGER_FL_CLEAR) {
+			tr->flags |= filter->trigger.flags; /* Remember what we remove */
```

Comment:
Oh, I meant if it needs to remember  the old flags.  I'm ok with the syntax that removes all actions at once.

But I'm a bit skeptical to change the existing trigger syntax.  As I said in other comment, `-T func@depth` can be confusing and/or resulting in misbehavior.  How about making `@clear` can take an argument.

* `-T func@clear` will remove all actions
* `-T func@clear=read+depth` will remove read and depth trigger actions only

Reply:
> I meant if it needs to remember the old flags.

OK my bad. This is useful when removing filters, to keep count of them.

> `-T func@depth` can be confusing and/or resulting in misbehavior. How about making `@clear` can take an argument.

I'm OK with this syntax, I'll use it. Otherwise I would have added a check at the end of `setup_trigger_action()` to invalidate e.g. the `depth` flag if no value is given, thus making `-T func@depth` illegal. Wdyt?